From 82cc7aca5fb7c7103256229a3ead0b8075da5813 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Wed, 5 Sep 2018 17:00:25 +0100
Subject: [PATCH] sunxi: DT: A64: add Pine64-LTS support

The Pine64-LTS board is a new board version of the Pine64, though
actually it's a non-SoM version of the SoPine and its baseboard.
In contrast to the original Pine64 it features LPDDR3 DRAM, an eMMC
socket and a SPI flash chip for storing firmware (like U-Boot).
Add it as a variant to the existing SoPine defconfig, since from a
software point of view they are really not different.

Signed-off-by: Andre Przywara <andre.przywara@arm.com>
Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 arch/arm/dts/Makefile                  |  1 +
 arch/arm/dts/sun50i-a64-pine64-lts.dts | 13 +++++++++++++
 configs/sopine_baseboard_defconfig     |  1 +
 3 files changed, 15 insertions(+)
 create mode 100644 arch/arm/dts/sun50i-a64-pine64-lts.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index ebfa227262..6cd440d6af 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -401,6 +401,7 @@ dtb-$(CONFIG_MACH_SUN50I) += \
 	sun50i-a64-nanopi-a64.dtb \
 	sun50i-a64-olinuxino.dtb \
 	sun50i-a64-orangepi-win.dtb \
+	sun50i-a64-pine64-lts.dtb \
 	sun50i-a64-pine64-plus.dtb \
 	sun50i-a64-pine64.dtb \
 	sun50i-a64-sopine-baseboard.dtb
diff --git a/arch/arm/dts/sun50i-a64-pine64-lts.dts b/arch/arm/dts/sun50i-a64-pine64-lts.dts
new file mode 100644
index 0000000000..72d6961dc3
--- /dev/null
+++ b/arch/arm/dts/sun50i-a64-pine64-lts.dts
@@ -0,0 +1,13 @@
+/*
+ * SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+ *
+ * Copyright (c) 2018 ARM Ltd.
+ */
+
+#include "sun50i-a64-sopine-baseboard.dts"
+
+/ {
+	model = "Pine64 LTS";
+	compatible = "pine64,pine64-lts", "allwinner,sun50i-r18",
+		     "allwinner,sun50i-a64";
+};
diff --git a/configs/sopine_baseboard_defconfig b/configs/sopine_baseboard_defconfig
index 9ede081c08..eead38a1e3 100644
--- a/configs/sopine_baseboard_defconfig
+++ b/configs/sopine_baseboard_defconfig
@@ -15,6 +15,7 @@ CONFIG_NR_DRAM_BANKS=1
 # CONFIG_SPL_DOS_PARTITION is not set
 # CONFIG_SPL_EFI_PARTITION is not set
 CONFIG_DEFAULT_DEVICE_TREE="sun50i-a64-sopine-baseboard"
+CONFIG_OF_LIST="sun50i-a64-sopine-baseboard sun50i-a64-pine64-lts"
 CONFIG_SUN8I_EMAC=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_SYS_USB_EVENT_POLL_VIA_INT_QUEUE=y
-- 
2.19.0.rc0

